youtube_dl/downloader/__init__.py

```diff
@@ -30,14 +32,17 @@ def get_suitable_downloader(info_dict, params={}):
     protocol = determine_protocol(info_dict)
     info_dict['protocol'] = protocol
 
+    if (info_dict.get('start_time') or info_dict.get('end_time')) and FFmpegFD.can_download(info_dict):
```

Comment:
i think this would be the correct behaviour.
the extractor that support all features needed is the one that will be selected(for now it's the only one that support this feature).